Content/Items/Misc/Accessories.TaintedGreataxe.cs

```diff
@@ -0,0 +1,469 @@
+                stickyAI = false;
+                Projectile.velocity = Vector2.UnitY;
+                EmbeddingCooldown = 120;
+            }
+
+            if (!stickyAI)
+            {
+                Projectile.rotation += Projectile.velocity.X * 0.025f;
+
+                Vector2 direction = targetNPC.Center - Projectile.Center;
+                if (direction.Length() > 25)
+                {
+                    direction.Normalize();
+                    Projectile.velocity = Vector2.Lerp(Projectile.velocity, direction * 35f, 0.02f);
+                }
+
+                Dust.NewDustPerfect(Projectile.Center, ModContent.DustType<Dusts.GlowFastDecelerate>(), null, 0, new Color(85, 220, 55), 0.35f);
+            }
+            else
+            {
+                Dust.NewDust(targetNPC.position, targetNPC.width, targetNPC.height, ModContent.DustType<Dusts.GlowFastDecelerate>(), 0, -2.5f, newColor: new Color(85, 220, 55), Scale: 0.4f);
+
+
+                if (Main.mouseRight)// right clicking the projectile unembeds it
+                    if (Main.myPlayer == Projectile.owner)
+                        if (Projectile.Distance(Main.MouseWorld) < 30f)
+                        {
+                            Embedding = false;
+                            stickyAI = false;
+                            Projectile.velocity = Vector2.UnitY;
+                            EmbeddingCooldown = 120;
+                        }
+            }
+        }
+
+        private void IdleMovement()
+        {
+            Projectile.rotation += Projectile.velocity.X * 0.04f;
+
+            //modified movement code from Diane Crecent
+            float speed = 15;
+            Vector2 direction = Owner.Center - Projectile.Center;
+            if (direction.Length() > 1500)
+            {
+                direction.Normalize();
+                Vector2 TeleportPos = Owner.Center + Main.rand.NextVector2Circular(100, 100);
+                Projectile.Center = TeleportPos;
+                for (int i = 0; i < 10; i++)
+                {
+                    Dust.NewDustPerfect(TeleportPos, ModContent.DustType<Dusts.GlowFastDecelerate>(), null, 0, new Color(200, 200, 205), 0.45f);
+
+                    Dust.NewDustPerfect(TeleportPos, ModContent.DustType<Dusts.GlowFastDecelerate>(), null, 0, new Color(85, 220, 55), 0.40f);
+                }
+            }
+
+            if (direction.Length() > 100)
+            {
+                direction.Normalize();
+                Projectile.velocity = Vector2.Lerp(Projectile.velocity, direction.RotatedByRandom(1.5f) * speed, 0.01f);
+            }
+            else if (Projectile.velocity == Vector2.Zero)
+                Projectile.velocity = Projectile.DirectionTo(Owner.Center);
+
+            Dust.NewDustPerfect(Projectile.Center, ModContent.DustType<Dusts.GlowFastDecelerate>(), null, 0, new Color(85, 220, 55), 0.35f);
+        }
+    }
+
+    public class TaintedGreataxeDust : ModDust
```

Comment:
Im partial to moving this dust out to a seperate file given the length of this class already, though its likely not explicitly needed.